raysect/core/math/function/vectorfunction2d/tests/test_base.py

```diff
@@ -0,0 +1,270 @@
+# Copyright (c) 2014-2020, Jack Lovell, Raysect Project
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+#     1. Redistributions of source code must retain the above copyright notice,
+#        this list of conditions and the following disclaimer.
+#
+#     2. Redistributions in binary form must reproduce the above copyright
+#        notice, this list of conditions and the following disclaimer in the
+#        documentation and/or other materials provided with the distribution.
+#
+#     3. Neither the name of the Raysect Project nor the names of its
+#        contributors may be used to endorse or promote products derived from
+#        this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+"""
+Unit tests for the Function2D class.
+"""
+
+import unittest
+from raysect.core.math import Vector3D
+from raysect.core.math.function.function2d.autowrap import PythonFunction2D
+from raysect.core.math.function.vectorfunction2d.autowrap import PythonVectorFunction2D
+
+# TODO: expand tests to cover the cython interface
+class TestFunction2D(unittest.TestCase):
+
+    def setUp(self):
+
+        self.refv1 = lambda x, y: Vector3D(10 * x + 5 * y, 10 * x - 5 * y, x + y)
+        self.refv2 = lambda x, y: Vector3D(x * x + y * y, x * x - y * y, x * y)
+
+        self.vf1 = PythonVectorFunction2D(self.refv1)
+        self.vf2 = PythonVectorFunction2D(self.refv2)
+
+        self.ref1 = lambda x, y: 10 * x + 5 * y
+        self.ref2 = lambda x, y: x * x + y * y
+
+        self.f1 = PythonFunction2D(self.ref1)
+        self.f2 = PythonFunction2D(self.ref2)
+
+    def test_call(self):
+        v = [-1e10, -7, -0.001, 0.0, 0.00003, 10, 2.3e49]
+        for x in v:
+            for y in v:
+                self.assertEqual(self.vf1(x, y), self.refv1(x, y), "Function2D call did not match reference function value.")
```

Comment:
All the unit test assert messages refer to Function2D, not VectorFunction2D.

Reply:
Thanks, I'll change them.